ambari-server/src/main/java/org/apache/ambari/server/state/svccomphost/ServiceComponentHostImpl.java

```diff
@@ -1043,6 +1048,14 @@ public void handleEvent(ServiceComponentHostEvent event)
           STOMPUpdatePublisher.publish(new HostComponentsUpdateEvent(Collections.singletonList(
               HostComponentUpdate.createHostComponentStatusUpdate(stateEntity, oldState))));
         }
+        if (event.getType().equals(ServiceComponentHostEventType.HOST_SVCCOMP_START)) {
+          HostComponentDesiredStateEntity desiredStateEntity = getDesiredStateEntity();
+          if (desiredStateEntity.getBlueprintProvisioningState() == BlueprintProvisioningState.IN_PROGRESS) {
```

Comment:
Any reason for double-check? The state goes from NONE -> IN_PROGRESS -> FININSHED, right?

Reply:
we want to update this field only with 1st start. No need for additional commit and update event after that.